src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,18 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (strcmp(arg, "map") == 0) {
```

Comment:
@liupan1111 I don't propose to modify global_init(). I propose to move it's call to do_map(). Right now we do this call so early just to filter out ceph common options, and it has turned out it does work well for our case: we still need to do preprocessing manually introducing potential bugs.

Consider a situation, when a user has a cluster named "map". And to make her life easier, she just added `--cluster map` to `CEPH_ARGS` env variable. Your current patch does not work as expected for this case.

My approach looks a little nice to me, still if Jason likes your patch, I will not have objection to merge it.